engine/apps/schedules/models/shift_swap_request.py

```diff
@@ -154,6 +156,76 @@ def web_link(self) -> str:
+
+        `09h00 - 17h00 (UTC) from Monday July 24, 2023 until July 28, 2023`
+
+        NOTE: For simplicity's sake, for shift swap requests that span multiple schedule events but either:
+        - don't start at the beginning of the first shift (ex. starts halfway into the first shift)
+        - don't end at the end of the last shift (ex. ends halfway into last shift)
+
+        We simply exclude this info from the summary. Here's an example.
+
+        Say we have a schedule for user A from 09h00 - 17h00 Monday - Friday. If user A opens a shift swap request
+        for 12h on Monday until 17h on Friday we would still show:
+
+        `09h00 - 17h00 (UTC) from Monday ... until Friday ...`
+
+        The same case occurs when the `swap_end` is earlier than the last shift's end time.
+        """
+        shifts = self.shifts()
+
+        def _format_time(dt: datetime.datetime) -> str:
+            return dt.strftime("%Hh%M")
+
+        def _format_date(dt: datetime.datetime) -> str:
+            return dt.strftime("%a %B %-d, %Y")
+
+        if not shifts:
+            # TODO: is it possible that this could ever be a legitimate case?
+            return ""
+
+        first_shift = shifts[0]
+        first_shift_start_datetime = first_shift["start"]
+        first_shift_end_datetime = first_shift["end"]
+        last_shift_start_datetime = first_shift["start"]
+        last_shift_end_datetime = first_shift["end"]
+
+        for shift in self.shifts():
+            shift_start = shift["start"]
+            shift_end = shift["end"]
+
+            if shift_start < first_shift_start_datetime:
+                first_shift_start_datetime = shift_start
+                first_shift_end_datetime = shift_end
+
+            if shift_end > last_shift_end_datetime:
+                last_shift_start_datetime = shift_start
+                last_shift_end_datetime = shift_end
+
+        if first_shift_start_datetime.date() == last_shift_end_datetime.date():
+            # all of the shift(s) occur on the same day
+            return f"{_format_time(first_shift_start_datetime)} - {_format_time(last_shift_end_datetime)} (UTC) on {_format_date(first_shift_start_datetime)}"
+
+        # the following two min/max usages are here to simplify the cases pertaining to a shift swap request
+        # that spans several events, but:
+        # - doesn't start at the beginning of the first shift (ex. starts halfway into the first shift)
+        # - doesn't end at the end of the last shift  (ex. ends halfway into last shift)
+        # see function docstring comment for more details
+        start_time = min([first_shift_start_datetime.time(), last_shift_start_datetime.time()])
+        end_time = max([first_shift_end_datetime.time(), last_shift_end_datetime.time()])
+
+        return f"{_format_time(start_time)} - {_format_time(end_time)} (UTC) from {_format_date(first_shift_start_datetime)} until {_format_date(last_shift_end_datetime)}"
+
```

Comment:
this is the main focus of this PR. See the function docstring for some caveats on what this `@property` does.

An alternative could be to _not include_ this `@property` and instead return the `shifts` array in the `ShiftSwapRequest` detail serializer and then have clients build their own summary from that.

**Pros of this ☝️**:
- 100% accurate. Handles tricky rotations + swaps that end/start at particular times

**Cons**:
- Each client has to have its own logic for parsing that array of events

@matiasb any opinion here?

Reply:
another alternative could be to have this `@property` return a longer, newline-delimited, string, such as this:
```
Monday July 24, 2023 9h00 - 17h00 (UTC)\nTuesday July 24, 2023 9h00 - 17h00 (UTC)\nWednesday July 25, 2023 9h00 - 17h00 (UTC)\nThursday July 26, 2023 9h00 - 17h00 (UTC)\nFriday July 27, 2023 9h00 - 17h00 (UTC)
```

which could then be easily used for something like this:
<img width="473" alt="Screenshot 2023-08-01 at 12 54 46" src="https://github.com/grafana/oncall/assets/9406895/3046cc5a-59d7-42d1-8979-476f094ed870">